### PR TITLE
sd-dhcp6-client: enum cleanups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 1]
```diff
@@ -97,19 +97,41 @@ static const uint16_t default_req_opts[] = {
...
+        [DHCP6_MESSAGE_RELAY_FORWORD]       = "relay-forword",
+        [DHCP6_MESSAGE_RELAY_REPLY]         = "relay-reply",
+        [DHCP6_MESSAGE_LEASE_QUERY]         = "lease-query",
+        [DHCP6_MESSAGE_LEASE_QUERY_REPLY]   = "lease-query-reply",
+        [DHCP6_MESSAGE_LEASE_QUERY_DONE]    = "lease-query-done",
+        [DHCP6_MESSAGE_LEASE_QUERY_DATA]    = "lease-query-data",
+        [DHCP6_MESSAGE_RECONFIGURE_REQUEST] = "reconfigure-request",
+        [DHCP6_MESSAGE_RECONFIGURE_REPLY]   = "reconfigure-reply",
+        [DHCP6_MESSAGE_DHCPV4_QUERY]        = "DHCPv4-query",
+        [DHCP6_MESSAGE_DHCPV4_RESPONSE]     = "DHCPv4-response",
```
Comment (poettering) [inline src/libsystemd-network/sd-dhcp6-client.c]: shouldn't thse be lowercase, too?

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 2]
```diff
@@ -1430,26 +1452,9 @@ static int client_receive_message(
...
-                return 0;
-
-        case DHCP6_ADVERTISE:
-        case DHCP6_REPLY:
-        case DHCP6_RECONFIGURE:
-                break;
-
-        default:
-                log_dhcp6_client(client, "Unknown message type %d", message->type);
+        if (!IN_SET(DHCP6_MESSAGE_ADVERTISE, DHCP6_MESSAGE_REPLY, DHCP6_MESSAGE_RECONFIGURE)) {
```
Comment (poettering) [inline src/libsystemd-network/sd-dhcp6-client.c]: this misses a `message->type` as first param

also, for a bunch of types this didn't generate any message, but now it does, that's not intended, is it?

File: src/libsystemd-network/sd-dhcp6-client.c (hunk 1 above)
Comment (yuwata) [reply src/libsystemd-network/sd-dhcp6-client.c]: This table is used only when logging, and not used parsing any options. So, `DHCP` should be better than `dhcp`, I think.

File: src/libsystemd-network/sd-dhcp6-client.c (hunk 2 above)
Comment (yuwata) [reply src/libsystemd-network/sd-dhcp6-client.c]: Yeah, fixed.

> also, for a bunch of types this didn't generate any message, but now it does, that's not intended, is it?

`log_dhcp6_client()` just logs in debug level. So, generating the log message in more cases should not cause any issues.

File: src/libsystemd-network/dhcp6-protocol.h [hunk 3]
```diff
@@ -71,8 +71,8 @@ enum {
         DHCP6_DECLINE                           = 9,
         DHCP6_RECONFIGURE                       = 10,
         DHCP6_INFORMATION_REQUEST               = 11,
-        DHCP6_RELAY_FORW                        = 12,
-        DHCP6_RELAY_REPL                        = 13,
+        DHCP6_RELAY_FORWORD                     = 12,
```
Comment (keszybz) [inline src/libsystemd-network/dhcp6-protocol.h]: `FORWARD`?

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 4]
```diff
@@ -97,19 +97,19 @@ static const uint16_t default_req_opts[] = {
...
+        [DHCP6_REQUEST]             = "request",
+        [DHCP6_CONFIRM]             = "confirm",
+        [DHCP6_RENEW]               = "renew",
+        [DHCP6_REBIND]              = "rebind",
+        [DHCP6_REPLY]               = "reply",
+        [DHCP6_RELEASE]             = "release",
+        [DHCP6_DECLINE]             = "decline",
+        [DHCP6_RECONFIGURE]         = "reconfigure",
+        [DHCP6_INFORMATION_REQUEST] = "information-request",
+        [DHCP6_RELAY_FORWORD]       = "relay-forword",
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp6-client.c]: Ditto.

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 5]
```diff
@@ -96,20 +96,42 @@ static const uint16_t default_req_opts[] = {
...
+        [DHCP6_MESSAGE_POOL_REQUEST]        = "pool-request",
+        [DHCP6_MESSAGE_POOL_RESPONSE]       = "pool-response",
+        [DHCP6_MESSAGE_UPDATE_REQUEST]      = "update-request",
+        [DHCP6_MESSAGE_UPDATE_REQUEST_ALL]  = "update-request-all",
+        [DHCP6_MESSAGE_UPDATE_DONE]         = "update-done",
+        [DHCP6_MESSAGE_CONNECT]             = "connect",
+        [DHCP6_MESSAGE_CONNECT_REPLY]       = "connect-reply",
+        [DHCP6_MESSAGE_DISCONNECT]          = "disconnect",
+        [DHCP6_MESSAGE_STATE]               = "state",
+        [DHCP6_MESSAGE_CONTACT]             = "contact",
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp6-client.c]: If those strings are only used for log messages, wouldn't it be better to spell them as "Binding Update", "Pool Request", "Update Request-All", "Update Done", etc.?

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 6]
```diff
@@ -1430,26 +1452,9 @@ static int client_receive_message(
...
-        case DHCP6_ADVERTISE:
-        case DHCP6_REPLY:
-        case DHCP6_RECONFIGURE:
-                break;
-
-        default:
-                log_dhcp6_client(client, "Unknown message type %d", message->type);
+        if (!IN_SET(message->type, DHCP6_MESSAGE_ADVERTISE, DHCP6_MESSAGE_REPLY, DHCP6_MESSAGE_RECONFIGURE)) {
+                log_dhcp6_client(client, "Received unexpected or unsupported message, ignoring: %s(%d)",
+                                 strna(dhcp6_message_type_to_string(message->type)), message->type);
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp6-client.c]:
```C
const char *type = dhcp6_message_type_to_string(message->type);
if (type)
        log_dhcp6_client(client, "Received unexpected %s message, ignoring.", type);
else
        log_dhcp6_client(client, "Received unsupported message code=%d, ignoring.", message->type);
```
?

File: src/libsystemd-network/dhcp6-protocol.h (hunk 3 above)
Comment (yuwata) [reply src/libsystemd-network/dhcp6-protocol.h]: Ouch!

**Comment by yuwata** [comment]
@keszybz Thank you for your review. Addressed all comments. Also, addressed an issue found by LGTM.com. Once dropping the green label. PTAL.

**Pull request merged**